### PR TITLE
[APB-3940][PT]: Add Inactive Trust Relationships and Refactor Inactive

### PR DESCRIPTION
**Pull request by JTY009**

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala [hunk 1]
```diff
@@ -236,30 +236,22 @@ class RelationshipsController @Inject()(
...
+      .cleanCopyStatusRecord(arn, mtdItId)
+      .map(_ => NoContent)
+      .recover {
+        case ex: RelationshipNotFound => NotFound(ex.getMessage)
       }
-    }
   }
 
-  def getInactiveVatRelationshipsAgent: Action[AnyContent] = Action.async { implicit request =>
+  //Generic
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala]: we can remove this comment - this doesn't tell anything important :)

File: app/uk/gov/hmrc/agentclientrelationships/model/ClientRelationship.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentclientrelationships.model
+import play.api.libs.json.{JsPath, Json, Reads}
+import uk.gov.hmrc.domain.SaAgentReference
+
+case class ClientRelationship(agents: Seq[Agent])
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientrelationships/model/ClientRelationship.scala]: this case class naming doesn't sound good to me - it just holds a list of Agents , no relationship info in it
could be `case class Agents(....)` ??

File: app/uk/gov/hmrc/agentclientrelationships/model/MtdItIdBusinessDetails.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentclientrelationships.model
+import play.api.libs.json.Json
+import uk.gov.hmrc.agentmtdidentifiers.model.MtdItId
+
+case class MtdItIdBusinessDetails(mtdbsa: MtdItId)
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientrelationships/model/MtdItIdBusinessDetails.scala]: i dont think we need a separate case class for wrapping mtdId ?

File: app/uk/gov/hmrc/agentclientrelationships/model/NinoBusinessDetails.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentclientrelationships.model
+import play.api.libs.json.Json
+import uk.gov.hmrc.domain.Nino
+
+case class NinoBusinessDetails(nino: Nino)
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientrelationships/model/NinoBusinessDetails.scala]: i dont think we need a separate case class for wrapping Nino ? same for below  `RegistrationRelationshipResponse` ?

File: app/uk/gov/hmrc/agentclientrelationships/services/FindRelationshipsService.scala [hunk 5]
```diff
@@ -31,27 +31,24 @@ import scala.concurrent.{ExecutionContext, Future}
...
   def getItsaRelationshipForClient(
-    nino: Nino)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[ItsaRelationship]] =
+    nino: Nino)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[ActiveRelationship]] =
     for {
       mtdItId       <- des.getMtdIdFor(nino)
-      relationships <- des.getActiveClientItsaRelationships(mtdItId)
+      relationships <- des.getActiveClientRelationships(mtdItId)
     } yield relationships
 
   def getVatRelationshipForClient(
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentclientrelationships/services/FindRelationshipsService.scala]: all the above 3 methods can be simplied with just one ?

File: app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala (hunk 1 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala]: ok

File: app/uk/gov/hmrc/agentclientrelationships/model/ClientRelationship.scala (hunk 2 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientrelationships/model/ClientRelationship.scala]: It was already there but can rename

File: app/uk/gov/hmrc/agentclientrelationships/model/MtdItIdBusinessDetails.scala (hunk 3 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientrelationships/model/MtdItIdBusinessDetails.scala]: same reason but can rearrange

File: app/uk/gov/hmrc/agentclientrelationships/model/NinoBusinessDetails.scala (hunk 4 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientrelationships/model/NinoBusinessDetails.scala]: same reason but can rearrange

File: app/uk/gov/hmrc/agentclientrelationships/services/FindRelationshipsService.scala (hunk 5 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientrelationships/services/FindRelationshipsService.scala]: another ticket but can rearrange for now

**Pull request merged**